src/providers/azure/mod.rs

```diff
@@ -179,7 +181,19 @@ impl Azure {
         .ok_or_else(|| "failed to get goal state: not found response".into())
     }
 
-    fn get_fabric_address() -> Result<IpAddr> {
+    fn get_fabric_address() -> IpAddr {
+        // try to fetch from dhcp, else use fallback; this is similar to what WALinuxAgent does
+        match Azure::get_fabric_address_from_dhcp() {
```

Comment:
Tip: this is an `unwrap_or_else(|| ...)`.

I don't have particular style preferences in case, so pick the form you prefer.

Reply:
Ahh of course, thanks. Used below! :arrow_down: